subsys/bluetooth/host/gatt.c

```diff
@@ -719,8 +723,44 @@ static void db_hash_gen(bool store)
 	atomic_set_bit(gatt_sc.flags, DB_HASH_VALID);
 }
 
+#if defined(CONFIG_BT_SETTINGS)
+static void sc_indicate(uint16_t start, uint16_t end);
+#endif
+
 static void db_hash_process(struct k_work *work)
 {
+#if defined(CONFIG_BT_SETTINGS)
+	if (atomic_test_and_clear_bit(gatt_sc.flags, DB_HASH_LOAD)) {
+		if (!atomic_test_bit(gatt_sc.flags, DB_HASH_VALID)) {
+			db_hash_gen(false);
+		}
+
+		/* Check if hash matches then skip SC update */
+		if (!memcmp(db_hash.stored_hash, db_hash.hash,
+			    sizeof(db_hash.stored_hash))) {
+			BT_DBG("Database Hash matches");
+			k_work_cancel_delayable(&gatt_sc.work);
```

Comment:
Can the cancel fail?  If so what happens?

Reply:
This would now be from a work handler so shouldn't fail in being canceled, right?

Reply:
If it can fail, and we need to do something, we should check the return value and do whatever. 

If you don't think we need to do anything, I think we should cast the return value to `(void)` to make it explicit.

Reply:
Yes, preemption of the work thread and SMP could both create a situation where the cancel could fail.

https://docs.zephyrproject.org/latest/reference/kernel/threads/workqueue.html#check-return-values shows recommended documentary practices.  [Casting to void](https://github.com/nrfconnect/sdk-nrf/pull/4520#discussion_r635263568) provides more background on the recommendation and why a cast to `(void)` is not useful.